src/Plugin/Filter/EntityEmbedFilter.php

```diff
@@ -143,7 +145,9 @@ public function process($text, $langcode) {
           watchdog_exception('entity_embed', $e);
         }
 
-        $this->setDomNodeContent($node, $entity_output);
+        // Ensure this element is using <div> now instead of <drupal-entity>.
+        $this->changeNodeName($node, 'div');
```

Comment:
I suggest changing node name only if the initial tag was `<drupal-entity>`. If the filter was used directly without WYSIWYG it might have wrapped in something other than `div` and we shouldn't change that I suppose.